client/stripe-utils/utils.js

```diff
@@ -455,10 +455,64 @@ export const getUpeSettings = () => {
 /**
  * Craft the defaultValues parameter, used to pre-fill
  * user email and phone number for Link in the Payment Element.
+ * On order pay and change payment method pages, also preloads all billing details
+ * from the customer billing data passed from the server.
  *
  * @return {Object} The defaultValues object for the Payment Element.
  */
 export const getDefaultValues = () => {
+	const stripeServerData = getStripeServerData();
+	const isOrderPay = stripeServerData?.isOrderPay;
+	const isChangingPayment = stripeServerData?.isChangingPayment;
+	const isAddPaymentMethod = stripeServerData?.isAddPaymentMethod;
+
+	// On order pay, change payment method, and add payment method pages, use billing data from customer.
+	if ( isOrderPay || isChangingPayment || isAddPaymentMethod ) {
+		const billingData = stripeServerData?.customerBillingData;
+
+		if ( billingData && billingData.email ) {
```

Comment:
Nit: Should we not validate that the email after running `trim()`? I believe `'  '` will evaluate as truthy.
```suggestion
		if ( billingData && billingData.email?.trim() ) {
```